### ISSUE TITLE
"experimental Nix feature 'ca-derivations' is disabled" even though it is enabled

### ISSUE THREAD

**Issue by samuela** [NixOS/nix#9090]
**Describe the bug**

Similar to https://github.com/NixOS/nix/issues/4784, but relating to `nix build` as opposed to `nixes-rebuild`.

I'm seeing 
```
error: experimental Nix feature 'ca-derivations' is disabled; use '--extra-experimental-features ca-derivations' to override
```
errors despite all efforts to enable the flag.

**Steps To Reproduce**

I have 
```
# ~/.config/nix/nix.confg
experimental-features = nix-command flakes ca-derivations
```
and yet I'm still seeing 
```
❯ nix develop                                             
error: experimental Nix feature 'ca-derivations' is disabled; use '--extra-experimental-features ca-derivations' to override
```
when attempting to build a flake. Setting the flag as suggested in the error does no better:
```
❯ nix --extra-experimental-features ca-derivations develop
error: experimental Nix feature 'ca-derivations' is disabled; use '--extra-experimental-features ca-derivations' to override
```

Checking the settings shows that it is clearly enabled:
```
❯ nix show-config | grep ca-                              
experimental-features = ca-derivations flakes nix-command
ssl-cert-file = /nix/var/nix/profiles/default/etc/ssl/certs/ca-bundle.crt
```

**Expected behavior**

ca-derivations configuration to work as expected.

**`nix-env --version` output**
```
❯ nix-env --version
nix-env (Nix) 2.15.1
```

(FWIW I ran the macOS multi-user upgrade directions in [these docs](https://nixos.org/manual/nix/stable/installation/upgrading), but my installation remained on 2.15.1.)

**Additional context**

I'm running on macOS 13.3.1, aarch64-darwin.

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by samuela** [comment]
Removing `ca-derivations` from my `~/.config/nix/nix.conf` makes the issue go away, but obv at the cost of losing content addressed features.

**Comment by abathur** [comment]
Have you restarted the daemon?

**Comment by samuela** [comment]
Yes, as part of the upgrade process: `sudo -i sh -c 'nix-channel --update && nix-env --install --attr nixpkgs.nix && launchctl remove org.nixos.nix-daemon && launchctl load /Library/LaunchDaemons/org.nixos.nix-daemon.plist'`

**Comment by abathur** [comment]
I'd try rebooting to be really sure, I guess. 

I don't remember where I saw it and haven't tried ca-derivations myself, but I do remember seeing someone say that this feature was ~different in that the daemon did need to get restarted after the feature was set before it would apply because of something something database format.

**Comment by Ericson2314** [comment]
Yes it does requires some new tables be added to the SQLite DB.

**Comment by samuela** [comment]
> @samuela I think reinstalling the plist file and restarting the service might be separate?

Tbh I have no idea. I've just been trying to follow the docs as closely as possible

**Comment by onsails** [comment]
seeing same issue on macos
